### DISCUSSION TITLE
what causes useAsyncData to run again on a static generated site, client side?

### DISCUSSION THREAD

**Discussion by pixelpaulaus** [nuxt/nuxt#29288] [Questions]
Hello all,
Using all the latest Nuxt version, I have noticed if I link to a page via a NuxtLink ( so there is a prefetch ) and that link has a #hash or ?query in the URL. That useAsyncData will fire again, client side, on a static generated site.
So can anyone help me figure out the reason that would cause it to fire again instead of using the payload ( which is valid ) which was created during the static generate.

**Comment by TheAlexLichter** [comment]
That can happen if the data wasn't generated during prerendering (e.g. crawler didn't find the site, different key etc. etc.)

**Reply by pixelpaulaus** [reply]
thats the strange thing, it is getting the data all okay. If I access the page directly, not via a nuxt link. it renders fine with all the data. This  only happens when accessing the page via a nuxtlink.

**Reply by TheAlexLichter** [reply]
Hm.. Can you check the payload file that should be loaded when navigating on the client-side (e.g. via NuxtLink)?

**Reply by pixelpaulaus** [reply]
the payload file is valid, nothing wrong with it. It just wants to get a new one for some reason, when the NuxtLink has the #hash or ?query in the Url.
The fact that the Static generated page is perfect and correct when visited directly, and no errors in console come up, also shows that it is correct.
And when I remove the #hash or ?query for the URL, it also loads perfect fine via a NuxtLink.
And there is no data being fetched that uses the #hash or ?query from the URL.
If I put :prefetch="false" on the NuxtLink, it works fine.

**Reply by TheAlexLichter** [reply]
That's strange. Would be good to have some kind of reproduction for that behavior!
